### ISSUE TITLE
'MATLAB_fields' attribute not used/supported

### ISSUE THREAD

**Issue by frejanordsiek** [frejanordsiek/hdf5storage#3]
The Attribute ‘MATLAB_fields’ is not currently set when writing data that should be imported into MATLAB as structures, and is ignored when reading data from file. This is because the h5py package cannot work with its format. If a structure with fields ‘a’ and ‘cd’ are saved, the Attribute looks like the following when using the h5dump utility:

```
ATTRIBUTE "MATLAB_fields" {
   DATATYPE  H5T_VLEN { H5T_STRING {
      STRSIZE 1;
      STRPAD H5T_STR_NULLTERM;
      CSET H5T_CSET_ASCII;
      CTYPE H5T_C_S1;
   }}
   DATASPACE  SIMPLE { ( 1 ) / ( 1 ) }
   DATA {
   (0): ("a"), ("c", "d")
   }
}
```

MATLAB doesn’t strictly require this field, but supporting it will help with reading/writing empty MATLAB structs. Would probably require writing a custom Cython or C function to fix this.

**Comment by frejanordsiek** [comment]
This issue will be resolved completely when support for h5py < 2.3 is dropped (Issue #53 ).

**Comment by frejanordsiek** [comment]
Resolved for the master branch now that h5py < 2.3 support has been dropped (Issue #53).
